File: backend/app/schemas/childminder.py
from datetime import time
from pydantic import BaseModel, EmailStr


class AvailabilityRead(BaseModel):
    id: int
    childminder_id: int
    weekday: str
    start_time: time
    end_time: time
    available_slots: int

    model_config = {"from_attributes": True}


class ChildminderRead(BaseModel):
    id: int
    full_name: str
    email: EmailStr
    location_zone: str
    latitude: float
    longitude: float
    max_capacity: int
    current_capacity: int
    earliest_start_time: time
    latest_end_time: time
    supports_special_needs: bool
    years_experience: int
    tags: str | None
    profile_summary: str | None
    cluster_label: str | None
    availabilities: list[AvailabilityRead] = []

    model_config = {"from_attributes": True}

File: frontend/src/types/childminder.ts
export type ChildminderAvailability = {
  id: number;
  childminder_id: number;
  weekday: string;
  start_time: string;
  end_time: string;
  available_slots: number;
};

export type ChildminderDetail = {
  id: number;
  full_name: string;
  email: string;
  location_zone: string;
  latitude: number;
  longitude: number;
  max_capacity: number;
  current_capacity: number;
  earliest_start_time: string;
  latest_end_time: string;
  supports_special_needs: boolean;
  years_experience: number;
  tags?: string | null;
  profile_summary?: string | null;
  cluster_label?: string | null;
  availabilities: ChildminderAvailability[];
};